internal/pkg/scalable/deployments.go

```diff
@@ -27,23 +33,75 @@ type deployment struct {
 	*appsv1.Deployment
 }
 
-// SetReplicas sets the amount of replicas on the resource. Changes won't be made on kubernetes until update() is called
-func (d deployment) SetReplicas(replicas int) {
+// setReplicas sets the amount of replicas on the resource. Changes won't be made on kubernetes until update() is called
+func (d deployment) setReplicas(replicas int) error {
+	if replicas > math.MaxInt32 || replicas < math.MinInt32 {
+		return fmt.Errorf("replicas value exceeds int32 bounds")
+	}
+
+	// #nosec G115
 	newReplicas := int32(replicas)
 	d.Spec.Replicas = &newReplicas
+	return nil
 }
 
-// GetCurrentReplicas gets the current amount of replicas of the resource
-func (d deployment) GetCurrentReplicas() (int, error) {
+// getCurrentReplicas gets the current amount of replicas of the resource
+func (d deployment) getCurrentReplicas() (int, error) {
 	replicas := d.Spec.Replicas
 	if replicas == nil {
 		return 0, errNoReplicasSpecified
 	}
 	return int(*d.Spec.Replicas), nil
 }
 
+// ScaleUp upscale the resource when the downscale period ends
+func (d deployment) ScaleUp() error {
+	currentReplicas, err := d.getCurrentReplicas()
+	if err != nil {
+		return fmt.Errorf("failed to get current replicas for workload: %w", err)
+	}
+	originalReplicas, err := getOriginalReplicas(d)
+	if err != nil {
+		return fmt.Errorf("failed to get original replicas for workload: %w", err)
+	}
+	if originalReplicas == values.Undefined {
+		slog.Debug("original replicas is not set, skipping", "workload", d.GetName(), "namespace", d.GetNamespace())
+		return nil
+	}
+	if originalReplicas == currentReplicas {
+		slog.Debug("workload is already at original replicas, skipping", "workload", d.GetName(), "namespace", d.GetNamespace())
+		return nil
+	}
```

Comment:
I'd actually remove all the things used only for debug in all of the scaling funcs. 

```suggestion
	originalReplicas, err := getOriginalReplicas(d)
	if err != nil {
		return fmt.Errorf("failed to get original replicas for workload: %w", err)
	}
	if originalReplicas == values.Undefined {
		slog.Debug("original replicas is not set, skipping", "workload", d.GetName(), "namespace", d.GetNamespace())
		return nil
	}
```